### PR TITLE
Add autoDownload option

### LINKED ISSUE #57: Throw error when cached Electron isn't found instead of downloading by default
I use a private build of Electron that I store on my local machine, and I use `electron-packager` to package my application. To use my private Electron, I set the `cache` option.

Debugging my build scripts was very cumbersome, because `electron-download` gave no indication of whether it could find the Electron build in the cache location or not. If it can't find Electron, it defaults to downloading public releases.

This is not what I want! I don't want the publicly available bits. I'd rather `electron-download` throw an error that I can catch and act on quickly.

electron-userland/electron-packager#187

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Fixes #57

This allows you to disable downloading missing files

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -183,8 +186,10 @@ class ElectronDownloader {
       if (exists && !this.force) {
         debug('zip exists', this.cachedZip)
         this.checkIfZipNeedsVerifying(cb)
-      } else {
+      } else if (this.opts.autoDownload) {
         this.ensureCacheDir(cb)
+      } else {
+        cb(new Error('File: "' + this.cachedZip + '" does not exist locally and autoDownload is false'))
```
Comment (malept) [inline lib/index.js]: Can use interpolation syntax here

File: lib/index.js [hunk 2]
```diff
@@ -307,6 +312,10 @@ class ElectronDownloader {
...
 
 module.exports = function download (opts, cb) {
-  const downloader = new ElectronDownloader(opts)
-  downloader.downloadIfNotCached(cb)
+  try {
+    const downloader = new ElectronDownloader(opts)
+    downloader.downloadIfNotCached(cb)
+  } catch (err) {
+    cb(err)
+  }
```
Comment (malept) [inline lib/index.js]: Oof. Would be nice if this module was ported to a Promise-based API.

Comment (MarshallOfSound) [reply lib/index.js]: Yes, yes it would 😆 #70

File: lib/index.js (hunk 1 above)
Comment (MarshallOfSound) [reply lib/index.js]: Does node 4 support that?

Comment (malept) [reply lib/index.js]: Yup! Electron Packager uses it everywhere.

**Pull request merged**